b2share/modules/records/triggers.py

```diff
@@ -52,7 +53,12 @@ def register_triggers(app):
     before_record_insert.connect(create_record_files_bucket)
     after_record_insert.connect(index_record)
 
+    # TODO(edima): replace this check with explicit permissions
     before_record_update.connect(check_record_immutable_fields)
+
+    # don't trust the user, reset record schema
+    #   from (possibly updated) community field
+    before_record_update.connect(set_record_schema)
```

Comment:
`set_record_schema ` sets the schema to the last version of a community schema. Thus setting it again would override an old schema version with a new one, potentially making the record invalid.